### ISSUE TITLE
Remove the source copy of FileSystemWatcher code

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#6875]
1. Once https://github.com/mono/mono/pull/12377 is merged, remove our source copy of FSW and bump minimum mono version.
2. Rewrite FileWatcherService to just use FileSystemWatcher directly, as PathTree is no longer needed.

> VS bug [#765539](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/765539)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-16] Comment by **xamarinc**



Fixed in version 8.0.0.1293 (master)



Author: Marius Ungureanu  
 Commit: 0dc11f89eb3ce7e440737672e225449cc3cb0f70 (mono/monodevelop)
